main_exercise/client/src/Components/TodoList/TodoListComponent.js

```diff
@@ -0,0 +1,58 @@
+import {useEffect, useMemo, useState} from 'react'
+import styles from './TodoList.module.css'
+import Loader from '../Loader/Loader'
+import FilterComponent from '../Filter/FilterComponent'
+import TodoItemConnector from "../TodoItem/TodoItemConnector";
+
+export default function TodoListComponent({
+                                              filterTodoList,
+                                              todoList,
+                                              filteredTodoList,
+                                              getTodoList,
+                                              deleteSelected
+                                          }) {
+    const selectedCheckboxArray = []
+    const [todos, setTodos] = useState([])
+    const [isLoading, setIsLoading] = useState(false)
+    const mapTodos = useMemo(() => {
+        return todos.map((todo) => {
+            return (
+               <TodoItemConnector key={todo.id} {...todo} selectedArray={selectedCheckboxArray}/>
+            )
+        })
+    })
+
+    useEffect(() => {
+        setIsLoading(true)
```

Comment:
why do you need setTodos? you can just used the todos from the state..
More than that, you can use filteredTodoList all the time, and if there's no filter, it will just be all the todos